### ISSUE TITLE
[FEATURE] Support additional discovery and configuration systems (Consul & Kubernetes)

### ISSUE THREAD

**Issue by PierreBesson** [jhipster/jhipster-registry#320]
##### **Overview of the issue**

See the related thread on the mailing list: https://groups.google.com/d/msg/jhipster-dev/DlF1PWnsXy0/5krUP1QdCAAJ

##### **Motivation for or Use Case**

Be able to use the registry UI when not using Eureka

**Comment by Abu1982** [comment]
Hi, I'd like to work on this issue.

I've reviewed the thread and confirmed there is no active PR for it.

**Planned approach:** Analyze and fix based on error signals in jhipster/jhipster-registry

I'll submit a focused PR once the fix is ready and locally verified.

**Comment by 218broadway-svg** [comment]
Hi, is this  bounty still open for an external PR? I see the issue is marked work-in-progress but there does not appear to be an active PR. If it is still open, I can investigate a focused implementation and share validation notes before opening a PR.

**Comment by chaudl113** [comment]
/attempt #320

**Implementation plan:**
1. Add Consul client dependency to the registry's build configuration
2. Create a ConsulServiceDiscovery class implementing the existing DiscoveryService interface
3. Add Kubernetes API integration for service discovery via k8s endpoints API
4. Create configuration profiles (consul, kubernetes) alongside existing eureka config
5. Update registry UI to display services from all configured backends
6. Add health indicators for Consul and Kubernetes connections

Starting with Consul integration as it has simpler API surface.

**Comment by malcom-alloc** [comment]
I reviewed the linked design thread and current `main`, along with the existing discovery PRs, especially #618. One acceptance detail is unclear: the original thread describes reusing the UI for discovered applications’ instances, metrics screens, and Swagger API, but current `MetricsService` and `HealthService` call the Registry-local `management/*` endpoints; #618 also explicitly leaves remote-instance monitoring and API proxying out of scope.

For #320, is provider-independent discovery with the existing Registry-local administration screens the intended acceptance target, or is viewing the discovered applications’ metrics/API documentation still required? If the latter remains part of this posted bounty, which existing discovery PR should that work build on? Clarifying this would help target the remaining work without another overlapping adapter implementation.
